packages/faceted-search/src/components/Badges/BadgeTags/BadgeTags.component.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import Badge from '@talend/react-components/lib/Badge';
+import { BadgeTagsForm } from './BadgeTagsForm.component';
+import { BadgeFaceted } from '../BadgeFaceted';
+import {
+	callbacksPropTypes,
+	operatorPropTypes,
+	operatorsPropTypes,
+} from '../../facetedSearch.propTypes';
+
+const getSelectBadgeLabel = (value, t) => {
+	const labelAll = t('FACETED_SEARCH_VALUE_ALL', { defaultValue: 'All' });
+	if (value) {
+		const checkedCheckboxes = value.filter(v => v.checked);
+		if (checkedCheckboxes.length > 3) {
+			return t('FACETED_SEARCH_VALUES_COUNT', {
+				count: checkedCheckboxes.length,
+				defaultValue: '{{count}} values',
+			});
+		} else if (!checkedCheckboxes.length) {
+			return labelAll;
+		}
+		return checkedCheckboxes.map(val => val.label);
+	}
+	return labelAll;
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export const BadgeTags = ({
+	id,
+	label,
+	initialOperatorOpened,
+	initialValueOpened,
+	operator,
+	operators,
+	size,
+	value,
+	category,
+	callbacks,
+	t,
+}) => {
+	const [tagsValues, setTagsValues] = useState([]);
+	const [isLoading, setIsLoading] = useState(true);
+	useEffect(() => {
+		if (!callbacks.getTags) {
+			console.warn("get tags callback is not defined, tags can't be fetch from server");
+		}
+
+		setIsLoading(true);
+		callbacks.getTags().then(data => {
+			setTagsValues(data.map(item => ({ id: item, label: item })));
+			setIsLoading(false);
```

Comment:
Set that in a finally, or in case or fetch error, it will stay in loading state forever.